test/compliance/password-grant-type_test.js

```diff
@@ -0,0 +1,232 @@
+      response.body.refresh_token.should.equal(token.refreshToken);
+      response.body.expires_in.should.be.a('number');
+      response.body.scope.should.equal(scope);
+
+      token.accessToken.should.be.a('string');
+      token.refreshToken.should.be.a('string');
+      token.accessTokenExpiresAt.should.be.a('date');
+      token.refreshTokenExpiresAt.should.be.a('date');
+      token.scope.should.equal(scope);
+
+      db.accessTokens.has(token.accessToken).should.equal(true);
+      db.refreshTokens.has(token.refreshToken).should.equal(true);
+    });
+
+    it ('Succesfull authorization and authentication', async function () {
+      const tokenRequest = createDefaultRequest();
+      const tokenResponse = new Response({});
+
+      const token = await auth.token(tokenRequest, tokenResponse, {});
+
+      const authenticationRequest = createRequest({
+        body: {},
+        headers: {
+          'Authorization': `Bearer ${token.accessToken}`
+        },
+        method: 'GET',
+        query: {}
+      });
+      const authenticationResponse = new Response({});
+
+      const authenticated = await auth.authenticate(
+        authenticationRequest,
+        authenticationResponse,
+        {});
+
+      authenticated.scope.should.equal(scope);
+      authenticated.user.should.be.an('object');
+      authenticated.client.should.be.an('object');
+    });
+
+    it ('Username missing', async function () {
+      const request = createDefaultRequest();
+      const response = new Response({});
+
+      delete request.body.username;
+
+      await auth.token(request, response, {})
+        .catch(err => {
+          err.name.should.equal('invalid_request');
+        });
+    });
+
+    it ('Password missing', async function () {
+      const request = createDefaultRequest();
+      const response = new Response({});
+
+      delete request.body.password;
+
+      await auth.token(request, response, {})
+        .catch(err => {
+          err.name.should.equal('invalid_request');
+        });
+    });
+
+    it ('Wrong username', async function () {
+      const request = createDefaultRequest();
+      const response = new Response({});
+
+      request.body.username = 'wrong';
+
+      await auth.token(request, response, {})
+        .catch(err => {
+          err.name.should.equal('invalid_grant');
+        });
+    });
+
+    it ('Wrong password', async function () {
+      const request = createDefaultRequest();
+      const response = new Response({});
+
+      request.body.password = 'wrong';
+
+      await auth.token(request, response, {})
+        .catch(err => {
+          err.name.should.equal('invalid_grant');
+        });
+    });
+
+    it ('Client not found', async function () {
+      const request = createDefaultRequest();
+      const response = new Response({});
+
+      request.headers.authorization = 'Basic ' + Buffer.from('wrong:wrong').toString('base64');
```

Comment:
Let's use a simple random hash like `wrong:${Math.round(Math.random() * 1000000).toString(16)}` so we don't have any issues with card-coded credential warnings

Reply:
Nice suggestion. 

The comment was already in a previous commit